src/tex/optimize.cpp

```diff
@@ -118,9 +131,11 @@ auto compute_optimization_steps(const Texture &file, const Settings &sets) noexc
 
     auto res = OptimizationSteps{};
 
+    // Check if conversion is a must.
     res.convert = conversion_required(file, sets);
-    // do not recompress if original is already compressed
-    if (sets.compress && res.best_format != info.format && !DirectX::IsCompressed(info.format))
+
+    // Do not compress the image if already compressed.
+    if (sets.compress && !DirectX::IsCompressed(info.format))
```

Comment:
We could possibly setup more output checks after the optimization is done to ensure stuff that should not be changed actually stays the same.

I don't think the condition served any purpose. `res.best_format` is `DXGI_FORMAT_UNKNOWN` by default, which will never be equal to any input texture format as far as I can tell.

Reply:
It did serve a purpose. Obviously, it was supposed to be evaluated after best format was computed

Reply:
I actually did that, before moving it back and removing the condition. I don't think CAO should recompress images with higher compression levels unless necessary (as in due to resizing which requires decompression with DxTex).

There's also the fact that the current version always decompresses a compressed input (which is not necessary if not resizing), which meant that this whole `if` was meaningless anyway (`sets.convert` could be `false`, yet since `optimize` function decompresses the input, it is converted anyway).

The reasoning why recompressing (if not required) does not make sense is that once you compress an image with i.e. BC1 or BC3, you have already incured a loss of quality. Decompression only restores the original format, not the quality. Compressing again with i.e. BC7 can't be better than the original compressed result, because the input for it is already "bad". The only real effect is increasing the size of the output file by ~2 times (not sure about the size difference exactly).